overlord/snapstate/snapmgr.go

```diff
@@ -510,6 +521,8 @@ func (m *SnapManager) ensureRefreshes() error {
 	}
 	chg.Set("snap-names", updated)
 	chg.Set("api-data", map[string]interface{}{"snap-names": updated})
+
+	m.nextRefresh = nil
```

Comment:
Sorry, I assumed this was also called on global refreshes ("snap refresh" alone), but looking at the terminology above, it makes sense that it's only called on "auto".